.circleci/config.yml

```diff
@@ -13,27 +13,54 @@ e2eTest: &e2eTest
           wget -q $(curl -sS https://api.github.com/repos/giantswarm/e2e-harness/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
           chmod +x ./e2e-harness
 
-    - run: ./e2e-harness localkube
+    - run:
+        name: Get e2ectl binary
+        command: |
+          wget -q $(curl -sS https://api.github.com/repos/giantswarm/e2ectl/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
+          chmod +x ./e2ectl
 
     - run:
-        # Project specific: Only disable kube-dns in this project,
-        # since it cannot run side-by-side with core-dns.
-        name: Disable minikube kube-dns addon
-        command: sudo ./minikube-linux-amd64 addons disable kube-dns
+        name: Get kubectl binary
+        command: |
+          curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl
+          chmod +x ./kubectl
 
-    - run: ./e2e-harness setup --remote=false
+    - run:
+        name: Init cluster
+        command: |
+          ./e2ectl cluster create
+          cp $(./e2ectl kubeconfig path) ${TEST_DIR}/kubeconfig
+          mkdir .e2e-harness
+          echo "existingCluster: true" >> .e2e-harness/config.yaml
+          echo "remoteCluster: false" >> .e2e-harness/config.yaml
 
     - run:
         name: Publish chart to CNR using a temporary channel for testing
         command: ./architect publish --pipeline=false --channels=${CIRCLE_SHA1}-${TEST_NAME}
 
-    - run: ./e2e-harness test --test-dir=${TEST_DIR}
+    - run:
+        # Project specific: Delete built-in kube-dns + CoreDNS in this project,
+        # since it cannot run side-by-side with one installed by the test.
+        name: Delete built-in core-dns
+        command: |
+          export KUBECONFIG=$(./e2ectl kubeconfig path)
+          ./kubectl delete service kube-dns -n kube-system
+          ./kubectl delete deployment coredns -n kube-system
+          ./kubectl delete configmap coredns -n kube-system
+          ./kubectl delete clusterrolebinding system:coredns
+          ./kubectl delete serviceaccount coredns -n kube-system
+          ./kubectl delete clusterrole system:coredns
+
+    - run:
+        name: Run test
+        command: |
+          E2E_KUBECONFIG=/e2e/kubeconfig ./e2e-harness test --test-dir=${TEST_DIR}
```

Comment:
What is missing to completely move away from e2e-harness now?

Reply:
e2e-harness does a lot of stuff - it is integration test library, a CLI tool and a runtime. This is step forward, in understanding and reduction of usage of e2e-harness as library & CLI tool, moves that into CircleCI configuration which we could turn to an reusable orb at some point.

AFAIU when running tests, in addition to KIND cluster container which is started now with e2ectl, there are two additional containers involved, e2e-harness as runtime + golang for compilation of integration test code.
I think e2e-harness container gets by e2e-harness CLI tool:
- "project" environment variables set/passed and
- host network enabled, in case k8s cluster is on the same host.
e2e-harness encodes also common setup and teardown steps (e.g. Helm tiller installation and removal), and of course actual running of the integration test binary in between.

At least some if not all of these steps could be scripted in CircleCI config.yaml, especially if not all options supported by e2e-harness (like shipyard use) are no longer needed. We'll know more as more e2e tests get migrated away from minikube.

Move from e2e-harness/minikube to e2ectl/kind alone might allows us as @rossf7 suggested (with some additional work on e2ectl and CircleCI script) to export logs and attach them to the build.

Reply:
Yes, e2e-harness does too many things.  This improves the situation a lot by removing the minikube / localkube provisioning part.

Stevo, this write up of what the `e2e-harness test` command does is very useful.

I'd like us to simplify as much as possible. The integration test is Go code and we have the E2E_KUBECONFIG env var with the kind kubeconfig. Do we actually need these 2 containers?

Rather than rabbit holing the existing code I'd like to see a clean slate implementation in a new package in `e2esetup`. But I think that should be the next iteration of this and we shouldn't delay removing minikube.

Reply:
I agree, we should simplify where possible, try to eliminate e2e-harness usage completely now that we started going in this new direction.

Before continuing with `e2e-harness test` breakdown / replacement, I plan to do the same migration of e2e-harness/minikube to e2ectl/kind for one more e2e integration test, one for https://github.com/giantswarm/kubernetes-nginx-ingress-controller to get a bit wider perspective, not to overfit the solution for single project only.

Reply:
@sslavic Nice I like that approach a lot and IC is a good candidate for the 2nd project to update.